tools/mcp/dotnet/AzureSDKDevToolsMCP/README.md

```diff
@@ -0,0 +1,68 @@
+# AzureSDKDevToolsMCP
+
+AzureSDKDevToolsMCP is a .NET 9.0 web application designed to integrate with the Model Context Protocol (MCP) framework. It provides tools for service team to use in GitHub Copilot agent to run various TypeSpec based operations.
+For e.g. TypeSpec validation
+
+## Features
+
+- **TypeSpec Validation**: Includes tools to validate TypeSpec API specifications.
+- **MCP Integration**: Leverages the Model Context Protocol for server communication.
+- **Environment Configurations**: Supports both development and production configurations.
+
+## Prerequisites
+
+- .NET 8.0 SDK
+- Node.js and npm (for TypeSpec validation)
+- GitHub client (for authentication)
+
+## Getting Started
+Compile and build the tools to generate static exe.
+
+1. Clone the repository:
+   ```
+   git clone https://github.com/azure/azure-sdk-tools.git
+   ```
+
+2. Navigate to the project directory:
+   ```
+   cd tools/mcp/AzureSDKDevTools
+   ```
+
+3. Build the project:
+   ```
+   dotnet build .
+   ```
+
+4. Generate the static executable:
+   ```
+   dotnet publish -c Release -r win-x64 --self-contained true /p:PublishSingleFile=true /p:GenerateBuildInfoConfigFile=false
+   ```
+
+5. Add an entry in VS Code MCP server settings for `AzureSDKDevTools` by copying below JSON into mcp.json.
+
+```
+{  
+    "servers": {
+      "Azure SDK Dev Tools": {
+        "type": "stdio",
+        "command": "<Path to exe>\\AzureSDKDSpecTools.exe",
+    }
+  }
+}
+```
```

Comment:
We'll probably want linux friendly instructions as well down the road, but this will all probably be paved over by the centralized mcp framework thing anyway.

Reply:
I am thinking of following the approach taken by Azure MCP to wrap the executable in npm package to handle different platforms. But planning to do this later in the stage once all functionalities are ready in the server.